File: iterableapi/src/main/java/com/iterable/iterableapi/IterableInAppMemoryStorage.java
package com.iterable.iterableapi;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import java.util.ArrayList;
import java.util.List;

class IterableInAppMemoryStorage implements IterableInAppStorage {
    private List<IterableInAppMessage> messages = new ArrayList<>();

    IterableInAppMemoryStorage() {

    }

    @NonNull
    @Override
    public synchronized List<IterableInAppMessage> getMessages() {
        return new ArrayList<>(messages);
    }

    @Nullable
    @Override
    public synchronized IterableInAppMessage getMessage(String messageId) {
        for (IterableInAppMessage message : messages) {
            if (message.getMessageId().equals(messageId)) {
                return message;
            }
        }
        return null;
    }

    @Override
    public synchronized void addMessage(@NonNull IterableInAppMessage message) {
        messages.add(message);
    }

    @Override
    public synchronized void removeMessage(@NonNull IterableInAppMessage message) {
        messages.remove(message);
    }

    @Override
    public void saveHTML(@NonNull String messageID, @NonNull String contentHTML) {

    }

    @Override
    public String getHTML(@NonNull String messageID) {
        return null;
    }

    @Override
    public void removeHTML(@NonNull String messageID) {

    }
}

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableInAppMemoryStorage.kt
package com.iterable.iterableapi

import androidx.annotation.NonNull
import androidx.annotation.Nullable

import java.util.ArrayList

internal class IterableInAppMemoryStorage : IterableInAppStorage {
    private val messages = ArrayList<IterableInAppMessage>()

    //region IterableInAppStorage interface implementation
    @NonNull
    @Synchronized
    override fun getMessages(): List<IterableInAppMessage> {
        return ArrayList(messages)
    }

    @Nullable
    @Synchronized
    override fun getMessage(messageId: String): IterableInAppMessage? {
        for (message in messages) {
            if (message.messageId == messageId) {
                return message
            }
        }
        return null
    }

    @Synchronized
    override fun addMessage(@NonNull message: IterableInAppMessage) {
        messages.add(message)
    }

    @Synchronized
    override fun removeMessage(@NonNull message: IterableInAppMessage) {
        messages.remove(message)
    }

    override fun saveHTML(@NonNull messageID: String, @NonNull contentHTML: String) {

    }

    override fun getHTML(@NonNull messageID: String): String? {
        return null
    }

    override fun removeHTML(@NonNull messageID: String) {

    }
    //endregion
}
